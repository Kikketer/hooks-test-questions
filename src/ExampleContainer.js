import React, { useEffect } from 'react'
import ExampleView from './ExampleView'

const ExampleContainer = ({ onSomeBusinessDone, iHaveMounted }) => {
  useEffect(() => {
    // An example of a "on mount" style effect that we may need to test
    // Grab data, do some initial logic whatever, but test that I call a prop
    // within this mount call
    iHaveMounted()
  }, [])

  const onDoSomeBussinessLogic = (eventData = {}) => {
    // Do something with the data perhaps
    eventData.whoa = true
    onSomeBusinessDone(eventData)
  }

  return <ExampleView onSomething={onDoSomeBussinessLogic} />
}

export default ExampleContainer
