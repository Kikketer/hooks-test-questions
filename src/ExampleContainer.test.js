import React from 'react'
import renderer, { act } from 'react-test-renderer'
import ExampleContainer from './ExampleContainer'

// I need to mock children since shallow doesn't run any effects
// Say I need to test a "mount" effect
jest.mock('./ExampleView', () => props => <div data-id="ExampleView" {...props} />)

// So tests should test incoming data and validate outcoming data
describe('ExampleContainer', () => {
  test('should call iHaveMounted on mount', () => {
    // This just tests to see if a prop is called on mount
    // In real life we may check some params sent back or validate that a listener
    // was engaged (and properly removed on unmount)
    const mounted = jest.fn()
    act(() => {
      renderer.create(<ExampleContainer iHaveMounted={mounted} />)
    })
    expect(mounted).toHaveBeenCalled()
  })

  test('when a child does something expect the prop function to have `whoa`', () => {
    const onSomething = jest.fn()
    act(() => {
      renderer.create(<ExampleContainer onSomething={onSomething} />)
    })
    // Somehow trigger the child's action... not sure how
    // Urm....
    expect(onSomething).toHaveBeenCalledWith({ clicked: true, whoa: true })
  })
})
