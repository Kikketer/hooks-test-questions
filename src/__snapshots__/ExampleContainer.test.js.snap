// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleContainer should match snapshot 1`] = `
<div
  data-id="ExampleView"
  onSomething={[Function]}
/>
`;
